scale-decode/src/lib.rs

```diff
@@ -272,17 +230,15 @@ impl<'a, T> FieldIter<'a> for T where T: Iterator<Item = Field<'a>> {}
 // rather than rely on auto conversion, if they care about also being able to impl `DecodeAsType`.
 pub trait IntoVisitor {
     /// The visitor type used to decode SCALE encoded bytes to `Self`.
-    type Visitor: for<'scale, 'info> visitor::Visitor<Value<'scale, 'info> = Self, Error = Error>;
+    type AnyVisitor<R: TypeResolver>: for<'scale, 'info> visitor::Visitor<
+        Value<'scale, 'info> = Self,
+        Error = Error,
+        TypeResolver = R,
+    >;
     /// A means of obtaining this visitor.
-    fn into_visitor() -> Self::Visitor;
+    fn into_visitor<R: TypeResolver>() -> Self::AnyVisitor<R>;
 }
 
-// In a few places, we need an empty path with a lifetime that outlives 'info,
-// so here's one that lives forever that we can use.
-#[doc(hidden)]
-pub static EMPTY_SCALE_INFO_PATH: &scale_info::Path<scale_info::form::PortableForm> =
-    &scale_info::Path { segments: Vec::new() };
```

Comment:
This was a bit of a hack in the first place. For now I've removed the ability to obtain the "path" to some composite type, because it's not something that's needed for decoding/encoding a type, and with the generic `TypeResolver` stuff it would just be an extra detail we need to try and provide from any implementation (and for our legacy type decoder I imagine it would be an unnecessary addition).

We don't provide the path to a Variant/Sequence type either anyway.